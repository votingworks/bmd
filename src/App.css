/* Allow selectors to be used multiple times in this file for grouping of concerns. */
/* stylelint-disable no-duplicate-selectors */

/* Reset */
html {
  box-sizing: border-box;
  line-height: 1;
  user-select: none;
}
*,
*::before,
*::after {
  box-sizing: inherit;
}

body {
  margin: 0;
}

table {
  border-collapse: collapse;
}

fieldset {
  margin: 0;
  border: none;
  padding: 0;
}

legend {
  display: block;
}

:link,
:visited {
  color: rgb(0, 0, 238);
}

:focus {
  outline: rgb(77, 144, 254) dashed 0.25rem;
}

/* Foundational */
html,
body,
#root {
  height: 100%;
  overflow: hidden;
}
@media print {
  html,
  body,
  #root {
    height: auto;
    overflow: visible;
  }
}

/* Typography */
html {
  background: #edeff0;
  color: #263238;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  font-size: 24px; /* Magic number, must synced with JavaScript global defaults */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
@media print {
  html {
    background: #ffffff;

    /* Adjust printed ballot font-size */
    font-size: 16px !important; /* stylelint-disable-line declaration-no-important */
  }
}

/* Accessibility Utilities */
.visually-hidden {
  position: absolute !important; /* stylelint-disable-line declaration-no-important */
  width: 1px;
  height: 1px;
  overflow: hidden;
  clip: rect(1px, 1px, 1px, 1px);
}
.visually-hidden a:focus,
.visually-hidden input:focus,
.visually-hidden button:focus {
  position: static;
  width: auto;
  height: auto;
}

/* Utilities */
.print-only {
  display: none;
}
@media print {
  .print-only {
    display: block;
  }
  .no-print {
    display: none;
  }
}

@media print {
  /* Chrome on Linux prints scrollbars */
  ::-webkit-scrollbar {
    display: none;
  }
}

/* stylelint-disable-next-line selector-class-pattern */
html body .cvox_indicator_container {
  /* Extra specificity/!important ensure this rule is followed. */
  display: none !important; /* stylelint-disable-line declaration-no-important */
}

/* Adjust printed ballot paper settings */
@page {
  margin: 0.375in;
  size: letter portrait;
}
