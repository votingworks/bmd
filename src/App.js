import React, { Component } from 'react';
import PrintProvider, { Print, NoPrint } from 'react-easy-print';
import './App.css';


import QrReader from 'react-qr-scanner';

import Ballot from './Ballot';
import PrintableBallot from './PrintableBallot';

var ELECTION = {
  id: "DEMO",
  title: "Demo Election",
  contests: [
    {
      id: "president",
      type: "plurality",
      name: "President",
      options: [
        "Minnie Mouse",
        "Mickey Mouse",
        "Donald Duck"
        ]
    },
    {
      id: "senator",
      type: "plurality",
      name: "Senator",
      options: [
        "Chad Hanging",
        "Lev Ermachine",
        "John Smith"
        ]
    }
  ]
};

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.election = ELECTION;
  }

  ballotReady(ballot) {
    this.setState({ballot: ballot});
  }

  print() {
    window.print();
  }

  handleScan(data) {
    if (data!=null) {
      console.log(data);
    }
  }

  handleError() {
  }

  render() {
    let mainContent, printableBallot;

    if (!this.election) {
      printableBallot = (<div></div>);

      const previewStyle = {
        height: 440,
        width: 620
      };

      mainContent = (
        <div align="center">
          <QrReader
            style={previewStyle}
            onError={this.handleError}
            onScan={this.handleScan}
          />
        </div>
      );
    } else {

      if (this.state.ballot) {
        printableBallot = (
          <PrintableBallot election={this.election} ballot={this.state.ballot} />
        );

        mainContent = (
          <div>
            <div style={{paddingLeft: "400px", paddingRight: "400px", paddingBottom: "50px"}}>{printableBallot}</div>
            <button onClick={this.print.bind(this)} style={{fontSize: "2.5em"}}>Print</button>
          </div>

        );
      } else {
        printableBallot = (<div></div>);
        mainContent = (
          <Ballot election={this.election} onBallotReady={this.ballotReady.bind(this)} />
        );
      }
    }


    return (
      <PrintProvider>
        <Print single printOnly name="ballot">
          {printableBallot}
        </Print>

        <NoPrint force>
          <div className="App">
            <img src="./vw-checkmark.png" className="App-logo" alt="logo" />
            {mainContent}
          </div>
      </NoPrint>
      </PrintProvider>
    );
  }
}

export default App;
