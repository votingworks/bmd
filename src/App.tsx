import React from 'react'
import { BrowserRouter, Route } from 'react-router-dom'

import 'normalize.css'
import './App.css'

import Screen from './components/Screen'

import AppRoot from './AppRoot'

const App = () => (
  <BrowserRouter>
    <Screen>
      <Route path="/" component={AppRoot} />
    </Screen>
  </BrowserRouter>
)

export default App
