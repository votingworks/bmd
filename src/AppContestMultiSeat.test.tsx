import React from 'react'
import { fireEvent, render, wait } from '@testing-library/react'
import fetchMock from 'fetch-mock'

import App from './App'

import { noCard, voterCard, advanceTimers } from '../test/helpers/smartcards'

import {
  countyCommissionersContest,
  setElectionInLocalStorage,
  setStateInLocalStorage,
} from '../test/helpers/election'

let currentCard = noCard
fetchMock.get('/card/read', () => JSON.stringify(currentCard))

jest.useFakeTimers()

beforeEach(() => {
  window.localStorage.clear()
  window.location.href = '/'
})

it(`Single Seat Contest`, async () => {
  // ====================== BEGIN CONTEST SETUP ====================== //

  setElectionInLocalStorage()
  setStateInLocalStorage()

  const { container, getByText, queryByText } = render(<App />)

  // Insert Voter Card
  currentCard = voterCard
  advanceTimers()

  // Go to Voting Instructions
  await wait(() => fireEvent.click(getByText('Get Started')))
  advanceTimers()

  // Go to First Contest
  fireEvent.click(getByText('Start Voting'))
  advanceTimers()

  // ====================== END CONTEST SETUP ====================== //

  const candidate0 = countyCommissionersContest.candidates[0]
  const candidate1 = countyCommissionersContest.candidates[1]
  const candidate2 = countyCommissionersContest.candidates[2]
  const candidate3 = countyCommissionersContest.candidates[3]
  const candidate4 = countyCommissionersContest.candidates[4]

  // Advance to multi-seat contest
  while (!queryByText(countyCommissionersContest.title)) {
    fireEvent.click(getByText('Next'))
    advanceTimers()
  }

  // Select 5 candidates for 4 seats
  fireEvent.click(getByText(candidate0.name))
  fireEvent.click(getByText(candidate1.name))
  fireEvent.click(getByText(candidate2.name))
  fireEvent.click(getByText(candidate3.name))
  fireEvent.click(getByText(candidate4.name))

  // Overvote modal is displayed
  getByText(
    `You may only select ${countyCommissionersContest.seats} candidates in this contest. To vote for ${candidate4.name}, you must first unselect selected candidates.`
  )

  // Capture styles of Single Candidate Contest
  expect(container.firstChild).toMatchSnapshot()

  // Go to Review Screen
  while (!queryByText('Review Your Selections')) {
    fireEvent.click(getByText('Next'))
    advanceTimers()
  }
  fireEvent.click(getByText('Review Selections'))
  advanceTimers()

  // Expect to see the first four selected candidates
  expect(getByText(candidate0.name)).toBeTruthy()
  expect(getByText(candidate1.name)).toBeTruthy()
  expect(getByText(candidate2.name)).toBeTruthy()
  expect(getByText(candidate3.name)).toBeTruthy()
})
