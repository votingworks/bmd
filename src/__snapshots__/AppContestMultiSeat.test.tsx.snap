// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Seat Contest 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
}

.c0:focus {
  outline: none;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgb(169,169,169);
  background: #455a64;
  padding: 0.25rem;
}

.c12 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c12 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.25rem;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  border-bottom: 1px solid rgb(169,169,169);
  background: #455a64;
  padding: 0.25rem;
}

.c16 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c16 > * {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0.25rem;
}

.c11 {
  position: relative;
  border-radius: 0.125rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  background: #FFFFFF;
  cursor: pointer;
  padding: 0.5rem 0.5rem 0.5rem 4rem;
  text-align: left;
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

.c11::before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  border-right: 1px solid;
  border-color: rgb(211,211,211);
  border-radius: 0.125rem 0 0 0.125rem;
  background: #FFFFFF;
  width: 3rem;
  text-align: center;
  color: #028099;
  font-size: 2rem;
  font-weight: 700;
  content: '';
}

.c9 {
  position: relative;
  border-radius: 0.125rem;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  background: #028099;
  cursor: pointer;
  padding: 0.5rem 0.5rem 0.5rem 4rem;
  text-align: left;
  color: #FFFFFF;
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

.c9::before {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  border-right: 1px solid;
  border-color: #028099;
  border-radius: 0.125rem 0 0 0.125rem;
  background: #FFFFFF;
  width: 3rem;
  text-align: center;
  color: #028099;
  font-size: 2rem;
  font-weight: 700;
  content: 'âœ“';
}

.c13 {
  border: none;
  border-radius: 0.25rem;
  box-sizing: border-box;
  background: rgb(71,167,75);
  cursor: pointer;
  padding: 0.75rem 1rem;
  line-height: 1.25;
  color: #FFFFFF;
  touch-action: manipulation;
}

.c14 {
  border: none;
  border-radius: 0.25rem;
  box-sizing: border-box;
  background: rgb(211,211,211);
  cursor: pointer;
  padding: 0.75rem 1rem;
  line-height: 1.25;
  color: black;
  touch-action: manipulation;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  width: 100%;
  max-width: 66ch;
  line-height: 1.2;
}

.c3 h1 {
  margin: 2rem 0 1rem;
  line-height: 1.1;
  font-size: 1.5rem;
}

.c3 h2 {
  margin: 1.5rem 0 0.75rem;
  font-size: 1.25rem;
}

.c3 h3,
.c3 p {
  margin-top: 1rem;
  margin-bottom: 1rem;
  font-size: 1rem;
}

.c3 h1 + h2 {
  margin-top: -0.75rem;
}

.c3 h1 + p,
.c3 h2 + p {
  margin-top: -0.75rem;
}

.c3 h3 + p {
  margin-top: -1rem;
}

.c3 > :first-child {
  margin-top: 0;
}

.c3 > :last-child {
  margin-bottom: 0;
}

.c3 dl {
  margin: 1rem 0;
}

.c10 {
  word-break: break-word;
}

.c2 {
  margin: 0 auto;
  width: 100%;
  max-width: 35rem;
  padding: 1rem 0.75rem 0.5rem;
}

.c4 {
  text-transform: uppercase;
  font-size: 0.85rem;
  font-weight: 600;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  overflow: auto;
}

.c5::before,
.c5::after {
  position: absolute;
  z-index: 1;
  width: 100%;
  height: 0.25rem;
  content: '';
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c5::before {
  top: 0;
  opacity: 0;
  background: linear-gradient( to bottom, rgb(177,186,190) 0%, transparent 100% );
}

.c5::after {
  bottom: 0;
  opacity: 0;
  background: linear-gradient( to bottom, transparent 0%, rgb(177,186,190) 100% );
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c7 {
  margin: 0 auto;
  width: 100%;
  max-width: 35rem;
  padding: 0.5rem 0.5rem 1rem;
}

.c8 {
  display: grid;
  grid-auto-rows: minmax(auto,1fr);
  grid-gap: 1.25rem;
}

.c15 {
  text-align: center;
  color: #FFFFFF;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

@media (min-width:480px) {
  .c12 > *:only-child {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
    margin: auto;
    min-width: 33.333%;
  }
}

@media print {
  .c12 {
    display: none;
  }
}

@media (min-width:480px) {
  .c16 > *:first-child {
    margin-left: auto;
  }
}

@media (min-width:480px) {
  .c16 > * {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
  }
}

@media (min-width:480px) {
  .c16 > *:only-child {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
    min-width: 33.333%;
  }
}

@media print {
  .c16 {
    display: none;
  }
}

@media (min-width:480px) {
  .c11 {
    padding: 1rem 1rem 1rem 4rem;
  }
}

@media (min-width:480px) {
  .c9 {
    padding: 1rem 1rem 1rem 4rem;
  }
}

@media print {
  .c1 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

@media (min-width:480px) {
  .c3 {
    line-height: 1.3;
  }
}

@media (min-width:720px) {
  .c2 {
    padding: 1rem 1.5rem 0.5rem;
    padding-left: 5rem;
  }
}

@media (min-width:720px) {
  .c7 {
    padding-right: 1rem;
    padding-left: 1rem;
  }
}

@media print {
  .c15 {
    color: #FFFFFF;
  }
}

<div
  class="c0"
  tabindex="-1"
>
  <main
    class="c1"
  >
    <div
      class="c2"
      id="contest-header"
    >
      <div
        class="c3"
        id="audiofocus"
      >
        <h1
          aria-label="County Commissioners."
        >
          <div
            class="c4"
          >
            Franklin County
          </div>
          County Commissioners
        </h1>
        <p
          aria-label="Vote for 4. You have selected 4. Use the down arrow to hear your options. Use the right arrow to move to the next contest."
        >
          <strong>
            Vote for 
            4
            .
          </strong>
           You have selected
           
          4
          .
        </p>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <button
              aria-label="Camille Argent, Federalist."
              class="c9"
              data-id="argent"
              data-selected="true"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Camille Argent
                  </strong>
                  <br />
                  Federalist
                </p>
              </div>
            </button>
            <button
              aria-label="Chloe Witherspoon-Smithson, Federalist."
              class="c9"
              data-id="witherspoonsmithson"
              data-selected="true"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Chloe Witherspoon-Smithson
                  </strong>
                  <br />
                  Federalist
                </p>
              </div>
            </button>
            <button
              aria-label="Clayton Bainbridge, Federalist."
              class="c9"
              data-id="bainbridge"
              data-selected="true"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Clayton Bainbridge
                  </strong>
                  <br />
                  Federalist
                </p>
              </div>
            </button>
            <button
              aria-label="Charlene Hennessey, People's."
              class="c9"
              data-id="hennessey"
              data-selected="true"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Charlene Hennessey
                  </strong>
                  <br />
                  People's
                </p>
              </div>
            </button>
            <button
              aria-label="Eric Savoy, People's."
              class="c11"
              data-id="savoy"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Eric Savoy
                  </strong>
                  <br />
                  People's
                </p>
              </div>
            </button>
            <button
              aria-label="Susan Tawa, People's."
              class="c11"
              data-id="tawa"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Susan Tawa
                  </strong>
                  <br />
                  People's
                </p>
              </div>
            </button>
            <button
              aria-label="Mary Tawa, People's."
              class="c11"
              data-id="tawa-mary"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Mary Tawa
                  </strong>
                  <br />
                  People's
                </p>
              </div>
            </button>
            <button
              aria-label="Damian Rangel, Liberty."
              class="c11"
              data-id="rangel"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Damian Rangel
                  </strong>
                  <br />
                  Liberty
                </p>
              </div>
            </button>
            <button
              aria-label="Valarie Altman, Constitution."
              class="c11"
              data-id="altman"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Valarie Altman
                  </strong>
                  <br />
                  Constitution
                </p>
              </div>
            </button>
            <button
              aria-label="Helen Moore, Constitution."
              class="c11"
              data-id="moore"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Helen Moore
                  </strong>
                  <br />
                  Constitution
                </p>
              </div>
            </button>
            <button
              aria-label="John White, Whig."
              class="c11"
              data-id="white"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    John White
                  </strong>
                  <br />
                  Whig
                </p>
              </div>
            </button>
            <button
              aria-label="Joe Schmidt, Whig."
              class="c11"
              data-id="schmidt"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Joe Schmidt
                  </strong>
                  <br />
                  Whig
                </p>
              </div>
            </button>
            <button
              aria-label="Joe Smith, Whig."
              class="c11"
              data-id="smith"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Joe Smith
                  </strong>
                  <br />
                  Whig
                </p>
              </div>
            </button>
            <button
              aria-label="Amanda Marracini, Labor."
              class="c11"
              data-id="marracini"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Amanda Marracini
                  </strong>
                  <br />
                  Labor
                </p>
              </div>
            </button>
            <button
              aria-label="Martin Schreiner, Independent."
              class="c11"
              data-id="schreiner"
              data-selected="false"
              role="option"
              type="button"
            >
              <div
                class="c3"
              >
                <p
                  class="c10"
                >
                  <strong>
                    Martin Schreiner
                  </strong>
                  <br />
                  Independent
                </p>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
  <nav
    class="c12"
  >
    <button
      class="c13"
      id="next"
      role="option"
      type="button"
    >
      Next
    </button>
    <button
      class="c14"
      id="previous"
      role="option"
      type="button"
    >
      Back
    </button>
    <p
      class="c15"
    >
      9
       of 
      20
    </p>
  </nav>
  <nav
    class="c16"
  >
    <div />
    <button
      class="c14"
      role="option"
      type="button"
    >
      Settings
    </button>
  </nav>
</div>
`;
